bolt-sidecar/src/onchain_registry.rs

```diff
@@ -0,0 +1,163 @@
+use std::str::FromStr;
+
+use alloy::{
+    contract::Error as ContractError,
+    primitives::{Address, Bytes, B256, B512},
+    providers::{ProviderBuilder, RootProvider},
+    sol,
+    sol_types::{Error as SolError, SolInterface},
+    transports::{http::Http, TransportError},
+};
+use ethereum_consensus::primitives::BlsPublicKey;
+use eyre::bail;
+use reqwest::{Client, Url};
+use reth_primitives::keccak256;
+use serde::Serialize;
+
+use BoltManagerContract::{BoltManagerContractErrors, BoltManagerContractInstance, ProposerStatus};
+
+use crate::config::chain::Chain;
+
+/// A wrapper over a BoltManagerContract that exposes various utility methods.
+#[derive(Debug, Clone)]
+pub struct BoltManager(BoltManagerContractInstance<Http<Client>, RootProvider<Http<Client>>>);
+
+impl BoltManager {
+    /// Returns the address of the canonical BoltManager contract for a given chain, if present
+    pub fn address(chain: Chain) -> Option<Address> {
+        match chain {
+            Chain::Holesky => Some(
+                Address::from_str("0x440202829b493F9FF43E730EB5e8379EEa3678CF")
+                    .expect("valid address"),
+            ),
+            _ => None,
+        }
+    }
```

Comment:
why did you choose to place this here instead of in config::Chain where we already have all the chain specific stuff? To me it seems like a more appropriate place, wdyt?

Reply:
Agreed

Reply:
Ok, I don't have a strong preference I first imagined this as an association Manager -> Chain rather than Chain -> Manager but the latter works well too